### PR TITLE
[Improvement] Adjust script structure

### PR DESCRIPTION
**Pull request by dreamerlin**
Add directories for scripts for `tools`, only leaving `train` and `test` related files

### CODE REVIEW

File: configs/localization/bsn/README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ You can use the following commands to inference a model.
 2. For PGM Inference
 
     ```shell
-    python tools/bsn_proposal_generation.py ${CONFIG_FILE} [--mode ${MODE}]
+    python tools/misc/bsn_proposal_generation.py ${CONFIG_FILE} [--mode ${MODE}]
```
Comment (innerlee) [inline configs/localization/bsn/README.md:74]: misc is not a good name

Comment (dreamerlin) [reply configs/localization/bsn/README.md:74]: Any good idea?

Comment (dreamerlin) [reply configs/localization/bsn/README.md:74]: Let's keep it as `misc` ?

Comment (kennymckormick) [reply configs/localization/bsn/README.md:74]: I think misc is OK.

**Review by kennymckormick — COMMENTED** [review]
Suggest not to place `feature_extraction` in  `misc`, it's an important feature like training or testing. Besides, the `dist_clip_feature_extraction.sh`  will not work now since the PYTHONPATH should also be altered.

**Review by kennymckormick — APPROVED** [review]

**Pull request merged**